src/words/environment/WordsEnvironment.java

```diff
@@ -138,12 +140,26 @@ public WordsObject getObject(String objectName) throws WordsRuntimeException {
 	 * Returns a collection of all objects.
 	 */
 	public Collection<WordsObject> getObjects() {
-		Collection<WordsObject> allObjects = new HashSet<WordsObject>();
-		for (HashMap<String,WordsObject> scopeObjects : gettableObjects) {
-			allObjects.addAll(scopeObjects.values());
+		try {
+			return getObjectsByClass("thing");
```

Comment:
Minor side comment: it would probably be good for us to define "thing" as a final constant somewhere called BASE_CLASS rather than using the string directly every time.